mmdet/models/utils/upsampler.py

```diff
@@ -0,0 +1,72 @@
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn import xavier_init
+
+from mmdet.ops.carafe import CARAFEPack
+
+
+class PixelShufflePack(nn.Module):
+
+    def __init__(self, in_channels, out_channels, scale_factor,
+                 upsample_kernel):
+        super(PixelShufflePack, self).__init__()
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.scale_factor = scale_factor
+        self.upsample_kernel = upsample_kernel
+        self.upsample_conv = nn.Conv2d(
+            self.in_channels,
+            self.out_channels * scale_factor * scale_factor,
+            self.upsample_kernel,
+            padding=(self.upsample_kernel - 1) // 2)
+        self.init_weights()
+
+    def init_weights(self):
+        xavier_init(self.upsample_conv, distribution='uniform')
+
+    def forward(self, x):
+        x = self.upsample_conv(x)
+        x = F.pixel_shuffle(x, self.scale_factor)
+        return x
+
+
+upsampler_cfg = {
+    # format: layer_type: (abbreviation, module)
+    'nearest': ('nearest', nn.Upsample),
+    'bilinear': ('bilinear', nn.Upsample),
+    'deconv': ('deconv', nn.ConvTranspose2d),
+    'pixel_shuffle': ('pixel_shuffle', PixelShufflePack),
+    'carafe': ('carafe', CARAFEPack)
+}
+
+
+def build_upsampler_layer(cfg, postfix=''):
```

Comment:
1. I suggest using a consistent term "upsample" instead of "upsample" and "upsampler".
2. `postfix` is unnecessary.